### PR TITLE
Workflow update for Drupal 10.4

### LINKED ISSUE #1151: Add test check of Edge with Drupal 10.4
## Is your feature request related to a problem? Please describe.
Drupal 10.3.x and 11.0.x will receive security coverage until June 2025, when Drupal 11.2.0 is released. Drupal 10.4 will be a maintenance minor, the first in Drupal history. Maintenance minors receive limited ongoing forward-compatibility and security fixes only. Maintenance minor releases of Drupal 10 will be provided until the release of Drupal 12 in 2026.

## Describe the solution you would like
Add test check for Drupal 10.4 on 4.x & 3.x

### PR DESCRIPTION
**Pull request by kedarkhaire**
Added Drupal 10.4 check in workflow
Fixes #1151

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -39,6 +39,7 @@ jobs:
           # Should update the following as the minimum supported version from Drupal.org
           - "10.2.x"
```
Comment (divya-intelli) [inline .github/workflows/php.yml]: We can remove Drupal 10.2 from testing, as it has reached end of life.

Comment (kedarkhaire) [reply .github/workflows/php.yml]: Okay, I will remove D10.2 tests for all repositories 👍🏼

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -112,7 +112,7 @@ jobs:
         composer update --with-all-dependencies
 
     - name: "Allow plugins and dev dependencies for Drupal 10.3"
-      if: ${{ matrix.drupal-core == '10.3.x' }}
+      if: ${{ matrix.drupal-core == '10.3.x' || matrix.drupal-core == '10.4.x' }}
```
Comment (divya-intelli) [inline .github/workflows/php.yml]: Is this condition still needed?

Comment (kedarkhaire) [reply .github/workflows/php.yml]: Updated condtions

**Review by divya-intelli — APPROVED** [review]
LGTM
